lib/active_resource/connection.rb

```diff
@@ -144,6 +150,14 @@ def handle_response(response)
           when 301, 302, 303, 307
             raise(Redirection.new(response))
           when 200...400
+            unless @response_array_key.blank?
+              begin
+                new_response_body = ActiveSupport::JSON.decode(response.body)[@response_array_key]
```

Comment:
This looks like in the wrong place. Why do we need to handle JSON here? Should not the Format take care of it like it do actually?

Reply:
@rafaelfranca I moved it to the JSON Format class, please check it.